build/proxies-scenarios.yml

```diff
@@ -46,24 +49,55 @@ parameters:
     - displayName: "100 KB"
       arguments: --variable path=/?s=102400 --property payload=102400
 
-- name: protocols
+- name: serverSchemes
+  type: object
+  default:
+    - displayName: http
+      arguments: --variable serverScheme=http --property serverScheme=http
+    - displayName: https
+      arguments: --variable serverScheme=https --property serverScheme=https
+
+- name: downstreamSchemes
   type: object
   default:
     - displayName: http
-      arguments: --variable protocol=http --property protocol=http
+      arguments: --variable downstreamScheme=http --property downstreamScheme=http
+    - displayName: https
+      arguments: --variable downstreamScheme=https --property downstreamScheme=https
+
+- name: serverHttpProtocols
```

Comment:
We don't need two "sets" here, for server and downstream, but just one called protocols, that will define both server and downstream protocols, and just the combinations we want. Then add `requireServerHttp1` ... that we can check in the condition more easily with `supportServerHttp1`.

I can change it if you want

Reply:
This way we just list and enumerate the combinations we want:
http - http
https - https
h2 - h2
https - http
h2 - http

Reply:
That's how I had it when I first submitted the PR, but you wanted me to split protocols into upstream and downstream protocols. https://github.com/aspnet/Benchmarks/pull/1587#discussion_r492933409. I warned you this would lead to a combinatorial explosion. Did seeing how complicated this made the conditions make you change your mind?

Reply:
No, I just wanted to save the two values in the properties. But here the loop can define the two values at the same time.

Reply:
Also, h2 doesn't necessarily imply https. To fully define the scenario it would have to be something like `HTTP/1.1 with no TLS - HTTP/1.1 with no TLS`, `HTTP/2 with TLS - HTTP/2 with TLS`, etc...

But I think prior-knowledge HTTP/2 is rare enough that we can skip non-TLS HTTP/2 scenarios.

Reply:
I realized there was a 3rd loop, right. So we can put everything in the same loop, and yes we don't want h2 without tls (h2c). I think "protocol" is sufficient, for both tiers, no need to mention it's with tls, and let's assume h2 is with it. worst case we can also use h2c to note it's without it (which will never happen). 
So http, https, h2, and define the combinations we measure with these.